### PR TITLE
fix: fix all lint errors shown in golangci-lint

### PR DESCRIPTION
**Pull request by hanshal101**
## Description
As discussed with @dr-housemd fixed all the linting errors shown by `golangci-lint run ./...` command.

## Note
- All the error which contains unused functions or variables are been commented out in this PR.
- Few changes were added in some conditional statements (test and review required on them)

### CODE REVIEW

**Comment by hanshal101** [comment]
Just one lint error exists, tbh don't know exactly what's going on there, NEED HELP !!!

**Comment by dr-housemd** [comment]
@hanshal101  you can use the `//nolint` directive on that one. 

Also, please modify GitHub action so that CI fails when linting fails.

File: pkg/builddocker/build.go [hunk 1]
```diff
@@ -135,31 +132,31 @@ func convertExportCfgToDockerfileCfg(env hcl2nix.OCIArtifact, platform string) d
...
-		if err != nil {
-			return "", err
-		}
-	} else if err != nil {
-		return "", err
-	}
-
-	return bsfDir, nil
-}
+// func generateRandomFilename() string {
```
Comment (dr-housemd) [inline pkg/builddocker/build.go:135]: we can remove this

File: pkg/sbom/sbom.go [hunk 2]
```diff
@@ -122,14 +124,30 @@ func parseLockfileToSBOMNodes(document *sbom.Document, appNode *sbom.Node, lf *h
 
 		document.NodeList.AddNode(&snode)
 		if pkg.Runtime {
-			document.NodeList.RelateNodeAtID(&snode, appNode.Id, sbom.Edge_runtimeDependency)
+			if err := document.NodeList.RelateNodeAtID(
+				&snode,
+				appNode.Id,
+				sbom.Edge_runtimeDependency,
+			); err != nil {
+				fmt.Fprintf(os.Stderr, "error : %s\n", err)
```
Comment (dr-housemd) [inline pkg/sbom/sbom.go:132]: we should not print/log from `pkg/` .

Comment (dr-housemd) [reply pkg/sbom/sbom.go:132]: For now, let's `continue` if we run into an error.

File: pkg/sbom/sbom.go [hunk 3]
```diff
@@ -154,10 +172,14 @@ func parseDotGraph(document *sbom.Document, appNode *sbom.Node, graph *gographvi
 			},
 		}
 		document.NodeList.AddNode(&snode)
-		document.NodeList.RelateNodeAtID(&snode, appNode.Id, sbom.Edge_contains)
+		if err := document.NodeList.RelateNodeAtID(
+			&snode,
+			appNode.Id,
+			sbom.Edge_contains,
+		); err != nil {
+			fmt.Fprintf(os.Stderr, "error : %s\n", err)
```
Comment (dr-housemd) [inline pkg/sbom/sbom.go:180]: same.. we shouldn't log/print..

File: cmd/search/model.go [hunk 4]
```diff
@@ -35,9 +35,9 @@ func convLPR2Items(packages *buildsafev1.ListPackagesResponse) []list.Item {
 	return items
 }
 
-type (
-	errMsg struct{ error }
-)
+// type (
```
Comment (dr-housemd) [inline cmd/search/model.go:38]: remove it if it isn't used.

File: pkg/builddocker/build.go [hunk 5]
```diff
@@ -134,32 +131,21 @@ func convertExportCfgToDockerfileCfg(env hcl2nix.OCIArtifact, platform string) d
...
-		if err != nil {
-			return "", err
-		}
-	} else if err != nil {
-		return "", err
-	}
-
-	return bsfDir, nil
-}
+// func createTempDir() (string, error) {
```
Comment (dr-housemd) [inline pkg/builddocker/build.go:134]: remove this

**Comment by hanshal101** [comment]
should I remove everything which is not used?

**Review by dr-housemd — APPROVED** [review]

**Review by dr-housemd — APPROVED** [review]

**Pull request merged**